### DISCUSSION TITLE
Support of X.X.X.X version format in xmake set_version()

### DISCUSSION THREAD

**Discussion by jfulem** [xmake-io/xmake#5885] [Q&A] [answered]
Is it possible to use this format of version in xmake: X.X.X.X ? When I use it then I got an error from my `config.h.in`
```
// version
#define VERSION "${VERSION}"
#define VERSION_MAJOR ${VERSION_MAJOR}
#define VERSION_MINOR ${VERSION_MINOR}
#define VERSION_ALTER ${VERSION_ALTER}
#define VERSION_BUILD ${VERSION_BUILD}
#define VERSION_BRANCH "${GIT_BRANCH}"
#define VERSION_COMMIT "${GIT_COMMIT}"
#define GIT_COMMIT_DATE "${GIT_COMMIT_DATE}"
```
**error: cannot get variable(VERSION_MAJOR) in config.h.in**

**Comment by waruqi** [comment]
https://github.com/xmake-io/xmake/blob/3276686e4d2356dab029d94ada8f780e0702889a/core/src/xmake/xmake.config.h.in#L6
https://github.com/xmake-io/xmake/blob/3276686e4d2356dab029d94ada8f780e0702889a/core/xmake.lua#L5

**Reply by jfulem** [reply]
Well it works when I use x.x.x but it does not when I use 4: x.x.x.x

For example 0.0.1 works but 0.0.0.1 does not.

**Comment by waruqi** [comment, marked as answer]
Xmake only supports the semantic versioning format. If you want to use four numbers, you can use the build number instead.
